projects/eslint-config/plugins/liferay/lib/rules/no-aui.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+const message = 'avoid using AUI, use vanilla Javascript instead';
+const MESSAGE_ARRAY = 'use the native Array class';
+const MESSAGE_DATE = 'use the native Date object';
+const MESSAGE_EACH = 'use Array methods to iterate over items';
+const MESSAGE_GETBODY =
+	"this is a shortcut to A.one('body'), use document.querySelector";
+const MESSAGE_IO = 'use the native Fetch API';
+const MESSAGE_OBJECT = 'use the native Object class';
+const MESSAGE_MERGE = 'use Object.assign';
+const MESSAGE_NODE =
+	'use native methods, mostly found inside the Document interface';
+const MESSAGE_SELECTOR = 'use document.querySelector instead';
+
+module.exports = {
+	create(context) {
+		return {
+			MemberExpression(node) {
+				const nodeObjectName = node.object.name;
+				const nodePropertyName = node.property.name;
+
+				if (nodeObjectName === 'A') {
+					let preciseMessage;
+
+					switch (nodePropertyName) {
+						case 'one' || 'all':
+							preciseMessage = MESSAGE_SELECTOR;
+							break;
+						case 'Array':
+							preciseMessage = MESSAGE_ARRAY;
+							break;
+						case 'Date':
+							preciseMessage = MESSAGE_DATE;
+							break;
+						case 'each':
+							preciseMessage = MESSAGE_EACH;
+							break;
+						case 'getBody':
+							preciseMessage = MESSAGE_GETBODY;
+							break;
+						case 'io':
+							preciseMessage = MESSAGE_IO;
+							break;
+						case 'merge':
+							preciseMessage = MESSAGE_MERGE;
+							break;
+						case 'Node':
+							preciseMessage = MESSAGE_NODE;
+							break;
+						case 'Object':
+							preciseMessage = MESSAGE_OBJECT;
+							break;
+						default:
+							preciseMessage = message;
+					}
```

Comment:
We shouldn't disallow all uses of AUI since there are cases that we don't have a replacement for yet.